### PR TITLE
Feature/#1943 add mediaobject to playlist

### LINKED ISSUE #1943: As a user, I want to be able to add all tracks of a CD to a playlist at once
### So that 
I don't have to add track by track.

### Items for Discussion
- Should this happen via some action on the Media Object page, like a helper button?
- Should this be per-section or per-media object?
- Should this also be available in _bulk actions_?

### Done looks like
-  A user has the ability to add all segments of a section to a playlist in one step. 
- The structural metadata is used to inform the clip sizes and names of playlist items.
- Per discussion: 07/27 The scope of this is **ONLY** for a top-level "add all" link. 
- The "add to" interface should be similar functionally to the current add to playlist.

### PR DESCRIPTION
**Pull request by bkeese**
Fixes #1943 

Adds new ability to easily add one or all sections of a media object to a playlist. Entire sections may be added as single playlist items, or if structural metadata is available, sections may be added as multiple playlist items based on that structure.

### CODE REVIEW

**Comment by cjcolvar** [comment]
Related to #1943

File: spec/controllers/media_objects_controller_spec.rb [hunk 1]
```diff
@@ -1034,4 +1034,67 @@
...
+  end
+
+  describe "#add_to_playlist" do
+    let(:media_object) { FactoryGirl.create(:media_object, :with_master_file, title: 'Test Item') }
+    let(:master_file_with_structure) { FactoryGirl.create(:master_file, :with_structure) }
+    let(:playlist) { FactoryGirl.create(:playlist) }
+
+    before do
+      login_as 'administrator'
+      media_object.ordered_master_files += [master_file_with_structure]
```
Comment (cjcolvar) [inline spec/controllers/media_objects_controller_spec.rb]: You can achieve this in the create without the need for this extra handling:
`let(:media_object) { FactoryGirl.create(:media_object, ordered_master_files: [FactoryGirl.create(:master_file), master_file_with_structure], title: 'Test Item') }`

**Comment by bkeese** [comment]
I went back and forth on which controller to use and landed on MediaObject because that's what we're really operating on here. I don't think this will be reusable for the other naming story, as that work will happen client-side.

**Comment by cjcolvar** [comment]
Looks good. We'll need to test with large items and see if we need to create a background job.

**Pull request merged**